### PR TITLE
Integrate a migration-detector middleware

### PR DESCRIPTION
**Pull request by matburt**
This attempts to detect if there are migrations in-progress and will
force display an interstitial page in the process that attempts to
load the index page every 10s until it succeeds.

This is only attached in production settings so the development
environment can proceed even if the migrations haven't been applied yet

connect #23

### CODE REVIEW

File: awx/main/middleware.py [hunk 1]
```diff
@@ -162,3 +164,12 @@ def process_request(self, request):
         if request.path_info != new_path:
             request.path = request.path.replace(request.path_info, new_path)
             request.path_info = new_path
+
+
+class MigrationRanCheckMiddleware(object):
+
+    def process_request(self, request):
+        executor = MigrationExecutor(connection)
+        plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
```
Comment (wwitzel3) [inline awx/main/middleware.py:173]: Hah, this is neat. Wouldn't of thought of that.

**Review by wwitzel3 — APPROVED** [review]

**Review by ryanpetrello — APPROVED** [review]
Neat idea :)

File: awx/main/middleware.py (hunk 1 above)
Comment (ryanpetrello) [reply awx/main/middleware.py:173]: Is this expensive to perform on each request?

Comment (matburt) [reply awx/main/middleware.py:173]: Not that I've seen, I ran through it a couple of times. Building the initial list of "What does this need to do" seems pretty small.

I'd totally rather have this be something that happens as part of the installer that swaps out the nginx config when the migrations are done but we've gone down the path of separating task execution from web service.

**Pull request merged**